Sources/ArcGISToolkit/Utility/CredentialInputView.swift

```diff
@@ -14,59 +14,145 @@
 
-/// A view that prompts a user to provide credentials. It can be configured to require either a username and
-/// password, or a password only.
-///
-/// The view is implemented as a wrapper for a UIKit `UIAlertController` because as of iOS 16,
-/// SwiftUI alerts don't support visible but disabled buttons.
-struct CredentialInputView: UIViewControllerRepresentable {
+extension View {
+    /// Presents user experiences for collecting credentials from the user.
+    /// - Parameters:
+    ///   - isPresented: A Boolean value indicating whether or not the view is displayed.
+    ///   - fields: The fields shown in the view.
+    ///   - message: Descriptive text that provides more details about the reason for the alert.
+    ///   - title: The title of the alert.
+    ///   - cancelAction: The cancel action.
+    ///   - continueAction: The continue action.
+    @ViewBuilder func credentialInput(
+        isPresented: Binding<Bool>,
+        fields: CredentialInputSheetView.Fields,
+        message: String,
+        title: String,
+        cancelAction: CredentialInputSheetView.Action,
+        continueAction: CredentialInputSheetView.Action
+    ) -> some View {
+        modifier(
+            CredentialInputModifier(
+                isPresented: isPresented,
+                fields: fields,
+                message: message,
+                title: title,
+                cancelAction: cancelAction,
+                continueAction: continueAction
+            )
+        )
+    }
+}
+
+struct CredentialInputSheetView_Previews: PreviewProvider {
+    static var previews: some View {
+        Text("foo")
+        .credentialInput(
+            isPresented: .constant(true),
+            fields: .usernamePassword,
+            message: "You must sign in to access 'arcgis.com'",
+            title: "Authentication Required",
+            cancelAction: .init(
+                title: "Cancel",
+                handler: { _, _ in
+                    
+                }
+            ),
+            continueAction: .init(
+                title: "Continue",
+                handler: { username, password in
+                }
+            )
+        )
+    }
+}
+
+/// A view modifier that prompts for credentials.
+struct CredentialInputModifier: ViewModifier {
+    
+    /// A Boolean value indicating whether or not the view is displayed.
+    var isPresented: Binding<Bool>
+    
+    /// The fields shown in the view.
+    let fields: CredentialInputSheetView.Fields
+    
+    /// Descriptive text that provides more details about the reason for the alert.
+    let message: String
+    
+    /// The title of the alert.
+    let title: String
+    
+    /// The cancel action.
+    let cancelAction: CredentialInputSheetView.Action
+    
+    /// The continue action.
+    let continueAction: CredentialInputSheetView.Action
+    
+    @ViewBuilder func body(content: Content) -> some View {
+        content
+            .sheet(isPresented: isPresented) {
```

Comment:
> There is a bug that I found that I need to fix: if you dismiss a sheet by swiping it doesn't cancel the challenge. I need to look into that.

Yes, I'd actually noticed this but forgot to mention it, sorry! Something like this may fix it. It's just unclear what the proper handler arguments would be.

```suggestion
            .sheet(isPresented: isPresented) {
                cancelAction.handler("", "")
            } content: {
```

Reply:
I thin the better thing to do is to prevent the interactive dismiss. We can make the user deal with the challenge explicitly.